Marlin/src/lcd/ultralcd.cpp

```diff
@@ -802,7 +802,9 @@ void MarlinUI::update() {
           card.release();
           if (old_sd_status != 2) {
             set_status_P(PSTR(MSG_SD_REMOVED));
-            if (!on_status_screen()) return_to_status();
+            #if ENABLED(ULTRALCD)
+              if (!on_status_screen()) return_to_status();
+            #endif
```

Comment:
This will never be true because the option name is "`ULTRA_LCD`." That's been replaced in code with `HAS_SPI_LCD`. But note, this whole block of code (all lines from 67 to 1284) is already wrapped in `HAS_SPI_LCD`. So… I think the right condition here is probably `HAS_LCD_MENU`.